Source/WebCore/dom/InternalObserverFilter.cpp

```diff
@@ -119,31 +119,28 @@ class InternalObserverFilter final : public InternalObserver {
         m_idx += 1;
 
         if (matches)
-            m_subscriber->next(value);
+            protectedSubscriber()->next(value);
     }
 
     void error(JSC::JSValue value) final
     {
-        m_subscriber->error(value);
+        protectedSubscriber()->error(value);
     }
 
     void complete() final
     {
         InternalObserver::complete();
-        m_subscriber->complete();
+        protectedSubscriber()->complete();
     }
 
     void visitAdditionalChildren(JSC::AbstractSlotVisitor& visitor) const final
     {
-        m_subscriber->visitAdditionalChildren(visitor);
-        m_predicate->visitJSFunction(visitor);
+        protectedSubscriber()->visitAdditionalChildren(visitor);
```

Comment:
This is not right. These functions are accessed from non-main thread concurrently to the main thread so we can't ref these objects.

Reply:
How can I fix this? I'll raise a fix pr immediately

Reply:
Is this as simple as:
```diff
- protectedSubscriber()->visitAdditionalChildren(visitor);
+ m_subscriber->visitAdditionalChildren(visitor);
```

Reply:
Yes, though I think you'll need to mark `m_subscriber` as `const` to let the static analyzer know it is safe not to have a smart pointer on the stack.

Reply:
With that, do we need to use the `protectedXYZ()` pattern?

Reply:
If the data member is a Ref and marked as `const` then in theory no, since there is no risk of the object getting destroyed while we're calling a member function on it. I don't know if our static analyzer implement this logic yet though (@rniwa would know). In any case, I would keep your `protectedFoo()` call removals specific to `visitAdditionalChildren()` for now.

Reply:
Static analyzer recognizes const member variables as safe now

Reply:
If/once https://github.com/WebKit/WebKit/pull/37812 lands—Ill raise a follow up to remove the `protectedXYZ` methods. 

Thank you for your patience and the help here folk, I sincerely apologize for this.